bigquery/bigquerystorage/src/main/java/com/example/bigquerystorage/StorageSample.java

```diff
@@ -0,0 +1,142 @@
+
+
+public class StorageSample {
+
+  /*
+   * SimpleRowReader handles deserialization of the Avro-encoded row blocks transmitted
+   * from the storage API using a generic datum decoder.
+   */
+  private static class SimpleRowReader {
+
+    private final DatumReader<GenericRecord> datumReader;
+
+    // Decoder object will be reused to avoid re-allocation and too much garbage collection.
+    private BinaryDecoder decoder = null;
+
+    // GenericRecord object will be reused.
+    private GenericRecord row = null;
+
+    public SimpleRowReader(Schema schema) {
+      Preconditions.checkNotNull(schema);
+      datumReader = new GenericDatumReader<>(schema);
+    }
+
+    /**
+     * Sample method for processing AVRO rows which only validates decoding.
+     *
+     * @param avroRows object returned from the ReadRowsResponse.
+     */
+    public void processRows(AvroRows avroRows) throws IOException {
+      decoder = DecoderFactory.get()
+          .binaryDecoder(avroRows.getSerializedBinaryRows().toByteArray(), decoder);
+
+      while (!decoder.isEnd()) {
+        // Reusing object row
+        row = datumReader.read(row, decoder);
+        System.out.println(row.toString());
+      }
+    }
+  }
+
+  public static void main(String... args) throws Exception {
+    // Sets your Google Cloud Platform project ID.
+    // String projectId = "YOUR_PROJECT_ID";
+    String projectId = args[0];
+
+    try (BigQueryStorageClient client = BigQueryStorageClient.create()) {
+      String parent = String.format("projects/%s", projectId);
+
+      // This example uses baby name data from the public datasets.
+      TableReference tableReference = TableReference.newBuilder()
+          .setProjectId("bigquery-public-data")
+          .setDatasetId("usa_names")
+          .setTableId("usa_1910_current")
+          .build();
+
+      // We specify the columns to be projected by adding them to the selected fields,
+      // and set a simple filter to restrict which rows are transmitted.
+      TableReadOptions options = TableReadOptions.newBuilder()
+          .addSelectedFields("name")
+          .addSelectedFields("number")
+          .addSelectedFields("state")
+          .setRowRestriction("state = \"WA\"")
+          .build();
+
+      // Construct the session.
+      CreateReadSessionRequest readSessionRequest = CreateReadSessionRequest.newBuilder()
+          .setParent(parent)
+          .setTableReference(tableReference)
+          .setReadOptions(options)
+          .setRequestedStreams(1)
+          .setFormat(DataFormat.AVRO)
+          .build();
+
+      ReadSession session = client.createReadSession(readSessionRequest);
+
+      SimpleRowReader reader = new SimpleRowReader(
+          new Schema.Parser().parse(session.getAvroSchema().getSchema()));
+
+      // Assert that there are streams available in the session.  If this fails, consider writing
+      // results of a query to a named table rather than consuming cached results directly.
```

Comment:
Since this can also fail due to normal empty tables, we should probably mention those first and then talk about the known issue.

Reply:
Updated comment to reflect the empty table case.